ts/src/profiler.ts

```diff
@@ -216,21 +279,20 @@ export class Profiler extends common.ServiceObject {
    * a profile and uploads it.
    *
    * @return - time, in ms, to wait before asking profiler server again about
-   * collecting another profile.
-   *
-   * TODO: implement backoff and retry. When error encountered in
-   * createProfile() should be retried when response indicates this request
-   * should be retried or with exponential backoff (up to one hour) if the
-   * response does not indicate when to retry this request.
+   * collecting another profile, or retryer specifying exponential backoff.
    */
   async collectProfile(): Promise<number> {
     let prof: RequestProfile;
     try {
       prof = await this.createProfile();
     } catch (err) {
       this.logger.error(`Failed to create profile: ${err}`);
```

Comment:
Do we log errors by default? Let's make sure we don't spam the output of the user's program. If we log errors, I think the current code will print out the error when the error is the server-instructed backoff error which is a normal condition and should not be visible to the user by default.

Reply:
I've switched things logged at the error level to being logged at the warn level, and when the err is a server-instructed backoff error, I switched to logging at the debug level.